app-backend/db/src/main/scala/GeoJsonUploadDao.scala

```diff
@@ -0,0 +1,173 @@
+    FROM
+  """ ++ tableF
+
+  def getUploadById(uploadId: UUID): ConnectionIO[Option[GeojsonUpload]] =
+    query.filter(uploadId).selectOption
+
+  def unsafeGetUploadById(uploadId: UUID): ConnectionIO[GeojsonUpload] =
+    query.filter(uploadId).select
+
+  def insert(
+      newUpload: GeojsonUpload.Create,
+      projectId: UUID,
+      layerId: UUID,
+      annotationGroupId: UUID,
+      user: User
+  ): ConnectionIO[GeojsonUpload] = {
+    val upload = newUpload.toGeojsonUpload(
+      user,
+      projectId,
+      layerId,
+      annotationGroupId
+    )
+    sql"""
+       INSERT INTO geojson_uploads
+         (id, created_at, created_by, modified_at,
+          upload_status, file_type, upload_type,
+          files, project_id, project_layer_id, annotation_group,
+          keep_files)
+       VALUES (
+         ${upload.id}, ${upload.createdAt}, ${upload.createdBy}, ${upload.modifiedAt},
+         ${upload.uploadStatus}, ${upload.fileType}, ${upload.uploadType},
+         ${upload.files}, ${upload.projectId}, ${upload.projectLayerId}, ${upload.annotationGroup},
+         ${upload.keepFiles}
+       )
+      """.update.withUniqueGeneratedKeys[GeojsonUpload](
+      "id",
+      "created_at",
+      "created_by",
+      "modified_at",
+      "upload_status",
+      "file_type",
+      "upload_type",
+      "files",
+      "project_id",
+      "project_layer_id",
+      "annotation_group",
+      "keep_files"
+    )
+  }
+
+  def update(
+      upload: GeojsonUpload,
+      projectId: UUID,
+      projectLayerId: UUID,
+      id: UUID,
+      user: User
+  ): ConnectionIO[Int] = {
+    val idFilter =
+      fr"id = ${id} AND project_id = $projectId AND project_layer_id = $projectLayerId"
+    val oldUploadIO = unsafeGetUploadById(id)
+    val recordUpdateIO = (sql"""
+       UPDATE geojson_uploads
+       SET
+          modified_at = NOW(),
+          upload_status = ${upload.uploadStatus}
+     """ ++ Fragments.whereAndOpt(Some(idFilter))).update.run
+    (for {
+      oldUpload <- oldUploadIO
+      newStatus <- upload.uploadStatus.pure[ConnectionIO]
+      nAffected <- recordUpdateIO
+      userPlatform <- UserDao.unsafeGetUserPlatform(user.id)
+    } yield (oldUpload, newStatus, nAffected, userPlatform)) flatMap {
+      case (
+          oldUpload: GeojsonUpload,
+          newStatus: UploadStatus,
+          nAffected: Int,
+          platform: Platform
+          ) => {
+        (
+          oldUpload.uploadStatus,
+          newStatus,
+          platform.publicSettings.emailIngestNotification,
+          user.getEmail
+        ) match {
+          case (_, _, _, "") | (_, _, false, _) => {
+            logger.info(
+              s"GeoJson upload complete, but user ${oldUpload.createdBy} or platform ${platform.name} has not requested email notifications"
+            )
+            nAffected.pure[ConnectionIO]
+          }
+          case (UploadStatus.Processing, UploadStatus.Failed, true, _) => {
+            logger.info(
```

Comment:
Nice, I hadn't considered email notifications

Reply:
We may need to go back and make a new email template if it gets confusing between scene and annotation uploads, but I think it'll be useful.